### PR TITLE
Replaced PVC with Synchronization Approach

### PR DESCRIPTION
**Pull request by BLasan**
**Changes**

- Replaced PVC with Synchronization approach
- Fixed StatefulSets Permissions

### CODE REVIEW

File: artifacts/api-manager-artifacts/pattern-1/wso2am-p1-apim-1-conf.yaml [hunk 1]
```diff
@@ -116,7 +131,7 @@ data:
     #expiry_time = "2m"
 
     [apim.analytics]
-    enable = true
+    enable = false
```
Comment (pubudu538) [inline artifacts/api-manager-artifacts/pattern-1/wso2am-p1-apim-1-conf.yaml]: Let's not disable this.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -324,8 +340,8 @@ func (c *Controller) syncHandler(key string) error {
 		workerDeploymentName := "wso2-am-analytics-worker-" + apimanager.Name
 		workerServiceName := "wso2-am-analytics-worker-svc"
 
-		synapseConfigsPVCName := "wso2am-p1-am-synapse-configs"
-		executionPlanPVCName := "wso2am-p1-am-execution-plans"
+		// synapseConfigsPVCName := "wso2am-p1-am-synapse-configs"
+		// executionPlanPVCName := "wso2am-p1-am-execution-plans"
 		mysqlPVCName := "wso2am-mysql"
```
Comment (pubudu538) [inline pkg/controller/controller.go:343]: Let's remove commented code.

**Review by pubudu538 — CHANGES_REQUESTED** [review]

**Review by pubudu538 — APPROVED** [review]

File: pkg/controller/pattern1/deployment.go [hunk 3]
```diff
@@ -195,7 +199,7 @@ func Apim2Deployment(apimanager *apimv1alpha1.APIManager, z *configvalues, num i
 	initContainers = append(initContainers, apim1InitContainer)
 
 	// appending the analytics-worker init container
-	initContainers = append(initContainers, getAnalyticsWorkerInitContainers())
+	//initContainers = append(initContainers, getAnalyticsWorkerInitContainers())
```
Comment (YasasRangika) [inline pkg/controller/pattern1/deployment.go:202]: Can we remove this code segment if we don't need this in the future?

Comment (BLasan) [reply pkg/controller/pattern1/deployment.go:202]: This line doesn't exist in the master branch

Comment (BLasan) [reply pkg/controller/pattern1/deployment.go:202]: Closing as the mentioned line is not in the master branch

**Pull request merged**